x/staking/keeper/msg_server.go

```diff
@@ -372,3 +455,553 @@ func (k msgServer) Undelegate(goCtx context.Context, msg *types.MsgUndelegate) (
 		CompletionTime: completionTime,
 	}, nil
 }
+
+// CancelUnbondingDelegation defines a method for canceling the unbonding delegation
+// and delegate back to the validator.
+func (k msgServer) CancelUnbondingDelegation(goCtx context.Context, msg *types.MsgCancelUnbondingDelegation) (*types.MsgCancelUnbondingDelegationResponse, error) {
```

Comment:
Either remove or document the reason `CancelUnbondingDelegation` is needed by the LSM.

Reply:
Added docs [here](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/69e6bb7a41eae0f4eaced4a049e172c0bbfdf4d8)

Reply:
I'm not sure that the fix for this issue https://github.com/cosmos/cosmos-sdk/issues/12931 was backported.

Reply:
Note that backporting this fix would require a change to ICS. See https://github.com/cosmos/interchain-security/issues/945. cc @MSalopek

Reply:
Backported [here](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/7d651a65b4065ecefadc757ac793567d04a31a31) (please double check me!)

Reply:
The migration seems to be missing. See `migrateUBDEntries`.

Reply:
@mpoke Can you point me to the migration? I don't see it anywhere in sdk 46

Reply:
This PR contains it https://github.com/cosmos/cosmos-sdk/pull/12967. See `store.go`. I assume that this is the fix that was backported.

Reply:
Added it [here](https://github.com/iqlusioninc/cosmos-sdk/pull/17/files#diff-6c5577886732344217f06949fa07049439b6bd341e9701500f80c99f92a6a5ebR63-R142).